### ISSUE TITLE
Enhance search capabilities for the qiime website

### ISSUE THREAD

**Issue by douginator2000** [qiime/qiime_web_app#111]
Add in additional methods for discovering sequence data in the website. Please post suggested criteria below:

**Comment by rob-knight** [comment]
I think the key criteria are:
- keywords in title
- keywords in abstract
- keywords in metadata
- presence/absence of metadata field
- PI
- other people associated with project (names or institutions)
- status (submitted, approved, metadata received, metadata finalized, samples received, sequenced, processed, available, published)
- submission date (as range, they won't know exactly)
- run date (as range as they won't know exactly)

Being able to show table that lets you sort by any of the above would also help a lot.

What would be nice is presence/absence of interesting named taxa, but that's possibly a separate type of search

Gail, Greg and Cathy (cc:ed) likely have additional suggestions for things that would have helped them directly to find analyses.

Thanks,
Rob

On Jan 24, 2013, at 12:57 PM, Doug Wendel <notifications@github.com<mailto:notifications@github.com>> wrote:

Add in additional methods for discovering sequence data in the website. Please post suggested criteria below:

**Comment by gregcaporaso** [comment]
It would also be useful to be able to search either a sequence (via BLAST or usearch) or an OTU ID (easier to develop) and find out what types of samples it shows up in, where types of samples were defined by some metadata category.

**Comment by douginator2000** [comment]
Greg, interesting ideas. Can you elaborate on how you'd foresee searching on a sequence via BLAST or usearch, and if either of those is preferable if we implemented just one?

Searching by OTU ID would be pretty easy to implement.

**Comment by wasade** [comment]
Blast or usearch against ref to get otu id?

On Jan 29, 2013, at 11:07, Doug Wendel notifications@github.com wrote:

Greg, interesting ideas. Can you elaborate on how you'd foresee searching
on a sequence via BLAST or usearch, and if either of those is preferable if
we implemented just one?

Searching by OTU ID would be pretty easy to implement.

—
Reply to this email directly or view it on
GitHubhttps://github.com/qiime/qiime_web_app/issues/111#issuecomment-12848557.

**Comment by gregcaporaso** [comment]
Yes, exactly. The user would have some sequence, and want to know what _types_ of samples it's found in. You'd BLAST/usearch it against our reference, and if it hits it would list the different sample types (with relative abundances in each), and link to the relevant studies. I think this is a lower priority feature than the ones that Rob listed, but is one thing I could see the QIIME DB being really useful for.

**Comment by wasade** [comment]
+1
